src/main/java/com/ishland/c2me/mixin/optimization/reduce_allocs/chunkserialization/MixinThreadedAnvilChunkStorage.java

```diff
@@ -0,0 +1,109 @@
+package com.ishland.c2me.mixin.optimization.reduce_allocs.chunkserialization;
+
+import com.ishland.c2me.common.optimization.chunkserialization.ChunkDataSerializer;
+import com.ishland.c2me.common.optimization.chunkserialization.NbtWriter;
+import com.mojang.datafixers.DataFixer;
+import com.mojang.datafixers.util.Either;
+import net.minecraft.nbt.NbtElement;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.server.world.ThreadedAnvilChunkStorage;
+import net.minecraft.structure.StructureStart;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.world.chunk.Chunk;
+import net.minecraft.world.chunk.ChunkStatus;
+import net.minecraft.world.poi.PointOfInterestStorage;
+import net.minecraft.world.storage.VersionedChunkStorage;
+import org.apache.logging.log4j.Logger;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+
+import java.io.DataOutputStream;
+import java.nio.file.Path;
+
+@Mixin(ThreadedAnvilChunkStorage.class)
+abstract public class MixinThreadedAnvilChunkStorage extends VersionedChunkStorage implements VersionedChunkStorageAccessor {
+    @Final
+    @Shadow
+    private static Logger LOGGER;
+
+    @Final
+    @Shadow
+    private PointOfInterestStorage pointOfInterestStorage;
+
+    @Final
+    @Shadow
+    ServerWorld world;
+
+    @Shadow
+    private native boolean isLevelChunk(ChunkPos chunkPos);
+
+    @Shadow
+    private native byte mark(ChunkPos chunkPos, ChunkStatus.ChunkType chunkType);
+
+    public MixinThreadedAnvilChunkStorage(Path directory, DataFixer dataFixer, boolean dsync) {
+        super(directory, dataFixer, dsync);
+    }
+
+    /**
+     * @author Kroppeb
+     * @reason Reduces allocations
+     */
+    @Overwrite()
+    private boolean save(Chunk chunk) {
+        // [VanillaCopy]
```

Comment:
this should be `// TODO [VanillaCopy]` for ease maintainability for new versions